File: src/main/java/org/hydev/mcpm/client/arguments/parsers/ListParser.java
package org.hydev.mcpm.client.arguments.parsers;

import net.sourceforge.argparse4j.inf.Namespace;
import net.sourceforge.argparse4j.inf.Subparser;
import org.hydev.mcpm.client.commands.entries.ListController;

import java.util.function.Consumer;

/**
 * Command parser for List command
 */
public record ListParser(ListController controller) implements CommandParser
{
    @Override
    public String name()
    {
        return "list";
    }

    @Override
    public String description()
    {
        return "List installed plugins";
    }

    @Override
    public void configure(Subparser parser)
    {
        parser.addArgument("type").choices("all", "manual", "outdated").setDefault("all").nargs("?");
    }

    @Override
    public void run(Namespace details, Consumer<String> log)
    {
        controller.listAll(details.getString("type"), log);
    }
}

File: src/main/java/org/hydev/mcpm/client/arguments/parsers/ListParser.kt
package org.hydev.mcpm.client.arguments.parsers

import net.sourceforge.argparse4j.inf.Namespace
import net.sourceforge.argparse4j.inf.Subparser
import org.hydev.mcpm.client.commands.controllers.ListController
import org.hydev.mcpm.client.interaction.ILogger

/**
 * Command parser for List command
 */

data class ListParser(val controller: ListController) : CommandParser
{
    override val name get() = "list"
    override val description get() = "List installed plugins"

    override fun configure(parser: Subparser)
    {
        parser.addArgument("type").choices("all", "manual", "automatic", "outdated").setDefault("all").nargs("?")
    }

    override suspend fun run(details: Namespace, log: ILogger)
    {
        controller.listAll(details.getString("type"), log)
    }
}
